yarn-project/sequencer-client/src/test/l2-block-publisher.test.ts

```diff
@@ -40,20 +85,120 @@ describe.skip('L1Publisher integration', () => {
         retryIntervalMs: 100,
       },
     );
-  });
+  }, 60_000);
 
-  it('publishes l2 block data to l1 rollup contract', async () => {
-    const blockNumber = await ethRpc.blockNumber();
-    await publisher.processL2Block(l2Block);
+  it('Build 2 blocks of 4 txs building on each other', async () => {
+    const stateInRollup_ = await rollup.methods.rollupStateHash().call();
+    expect(hexStringToBuffer(stateInRollup_.toString())).toEqual(Buffer.alloc(32, 0));
 
-    const logs = await rollup.getLogs('L2BlockProcessed', { fromBlock: blockNumber });
-    expect(logs).toHaveLength(1);
-    expect(logs[0].args.blockNum).toEqual(42n);
+    for (let i = 0; i < 2; i++) {
+      const tx = await makeProcessedTx(
+        Tx.createPrivate(makeKernelPublicInputs(1 + i), emptyProof, makeEmptyUnverifiedData()),
+      );
 
-    const tx = await ethRpc.getTransactionByHash(logs[0].transactionHash!);
-    const expectedData = rollup.methods.process(l2Proof, l2Block.encode()).encodeABI();
-    expect(tx.input).toEqual(expectedData);
-  });
+      const txsLeft = [tx, await makeEmptyProcessedTx()];
+      const txsRight = [await makeEmptyProcessedTx(), await makeEmptyProcessedTx()];
+
+      // Set tree roots to proper values in the tx
+      await setTxHistoricTreeRoots(tx);
+
+      // Calculate what would be the tree roots after the txs from the first base rollup land and update mock circuit output
+      await updateExpectedTreesFromTxs(txsLeft);
```

Comment:
this could be moved into a helper function as it shows up alot in other sequencer tests